### ISSUE TITLE
Error: issue loading your offline database (with Firefox) (solutions available)

### ISSUE THREAD

**Issue by alfem** [standardnotes/web#166]
A few days ago the web app (https://app.standardnotes.org/) started to show this message:

_There was an issue loading your offline database. This could happen for two reasons:

1. You're in a private window in your browser. We can't save your data without access to the local database. Please use a non-private window.

2. You have two windows of the app open at the same time. Please close any other app instances and reload the page._

I have closed/reopen my Firefox (52.5.0), logout/login,  but nothing helps.

In console I can see these messages:

Offline database issue: error { target: IDBOpenDBRequest, isTrusted: true, eventPhase: 0, bubbles: true, cancelable: true, defaultPrevented: false, composed: false, timeStamp: 1511257544068041, cancelBubble: false, originalTarget: IDBOpenDBRequest, explicitOriginalTarget: IDBOpenDBRequest }  compiled.min-ad250c9c0d26de39e6090f2320894d7b602e41d668d5dd1b9ec1b5038ef9eb66.js:10:8857
Offline database issue: error { target: IDBOpenDBRequest, isTrusted: true, eventPhase: 0, bubbles: true, cancelable: true, defaultPrevented: false, composed: false, timeStamp: 1511257544061824, cancelBubble: false, originalTarget: IDBOpenDBRequest, explicitOriginalTarget: IDBOpenDBRequest }  compiled.min-ad250c9c0d26de39e6090f2320894d7b602e41d668d5dd1b9ec1b5038ef9eb66.js:10:8857

**Comment by mobitar** [comment]
Are you using Firefox with any sort of modified privacy behaviors? For example if you tell Firefox not to save browsing history, you'll see this message, because Firefox disables database access with this setting enabled. You'll also see this in a private session.

You're saying this used to work but it doesn't now?

**Comment by alfem** [comment]
Yes. I have been testing Standard Notes for weeks without this issue. 

In my organization we use ESR versions of Firefox because they do not change as fast as the other ones. But I do not have any special privacy settings.

**Comment by poisonborz** [comment]
**Important note if you are using Firefox**: I got this error, even with all the extensions disabled/most config options being default - but newly downloaded app seemed to work. Turns out improper shutdowns can corrupt (?) your local database.

Go to your profile folder and delete storage.sqlite / webappstore.sqlite files, and the issue is gone.

**Comment by alfem** [comment]
Some of those files are pretty huge. I am not sure about just deleting them. 

How can I use the new app?

**Comment by alfem** [comment]
I finally created another Firefox profile and Standard Notes works again. 

Probably Firefox local storage got corrupted as  @poisonborz said.

**Comment by TheHobbit81** [comment]
@alfem Do you mean a fresh install of Firefox? I have a similar issue and so far have not been able to remedy it.

**Comment by alfem** [comment]
No, I don't. It is my 'old and safe Extended Support Release' install, updated for months. I just started with --ProfileManager parameter and then created a new profile. 

With this new profile StandartNotes works fine. Finally I copied my bookmarks, certs and plugins from my old profile and now everything goes smooth (even better than before!). I guess a corrupted db was also the culprit of some lag when opening new tabs.
